src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/RiskService.kt

```diff
@@ -23,27 +25,122 @@ class RiskService(private val assessmentClient: AssessmentApiRestClient) {
     val sectionsAnswers = assessmentClient.getRoshSectionsForCompletedLastYearAssessment(crn)
     log.info("Section answers for crn $crn number of sections : ${sectionsAnswers?.sections?.size}")
     return AllRoshRiskDto(
-      RoshRiskToSelfDto(
-        RiskDto(ResponseDto.DK, "Previous concerns", "Current concerns"),
-        RiskDto(ResponseDto.NO, "Previous concerns", "Current concerns"),
-        RiskDto(ResponseDto.YES, "Previous concerns", "Current concerns"),
-        RiskDto(ResponseDto.DK, "Previous concerns", "Current concerns"),
-        RiskDto(null, null, null),
+      roshRiskToSelfDto(sectionsAnswers),
+      sectionsAnswers.toOtherRoshRisksDto(),
+      sectionsAnswers.toRiskRoshSummaryDto()
+    )
+  }
+
+  fun getOtherRoshRisk(crn: String): OtherRoshRisksDto {
+    log.info("Get Other Rosh Risk for crn $crn")
+    val sectionsAnswers = assessmentClient.getRoshSectionsForCompletedLastYearAssessment(crn)
+    log.info("Section answers for crn $crn number of sections : ${sectionsAnswers?.sections?.size}")
+    return sectionsAnswers.toOtherRoshRisksDto()
+  }
+
+  fun getRoshRisksToSelfByCrn(crn: String): RoshRiskToSelfDto {
+    log.info("Get Rosh Risk to Self for crn $crn")
+    val sectionsAnswers = assessmentClient.getRoshSectionsForCompletedLastYearAssessment(crn)
+    log.info("Section answers for crn $crn number of sections : ${sectionsAnswers?.sections?.size}")
+
+    return roshRiskToSelfDto(sectionsAnswers)
+  }
+
+  fun getRoshRiskSummaryByCrn(crn: String): RiskRoshSummaryDto {
+    log.info("Get Rosh Risk summary for crn $crn")
+    val sectionsAnswers = assessmentClient.getRoshSectionsForCompletedLastYearAssessment(crn)
+    log.info("Section answers for crn $crn number of sections : ${sectionsAnswers?.sections?.size}")
+    return sectionsAnswers.toRiskRoshSummaryDto()
+  }
+
+  private fun SectionAnswersDto?.toRiskRoshSummaryDto(): RiskRoshSummaryDto {
+    val roshSumAnswers = this?.sections?.get(SectionHeader.ROSH_SUMMARY.value)
+
+    return RiskRoshSummaryDto(
+      findAnswer(roshSumAnswers, "SUM1")?.freeFormText,
+      findAnswer(roshSumAnswers, "SUM2")?.freeFormText,
+      findAnswer(roshSumAnswers, "SUM3")?.freeFormText,
+      findAnswer(roshSumAnswers, "SUM4")?.freeFormText,
+      findAnswer(roshSumAnswers, "SUM5")?.freeFormText,
+      roshSumAnswers.toRiskInCommunity(),
+      roshSumAnswers.toRiskInCustody()
+    )
+  }
+
+  private fun Collection<QuestionAnswerDto>?.toRiskInCustody(): Map<RiskLevel, List<String>> {
+    val children = RiskLevel.fromString(findAnswer(this, "SUM6.1.2")?.staticText)
```

Comment:
what's the suggestion of where to extract the question codes constants? should we have a dto where we do list one by one naming them with the information we are extracting? i.e val childrenRiskInCustody = "SUM6.1.2"